src/linker/Linker/Driver.cs

```diff
@@ -276,6 +278,12 @@ public bool Run (ILogger customLogger = null)
 
 							continue;
 
+						case "--custom-assembly":
```

Comment:
Please instead of introducing option which makes sense only when used with `--custom-step` please extend `--custom-step` to take optional assembly path value

Reply:
I've changed the way we get the custom assemblies to the other option that @vitek-karas  originally suggested (use `,,` as a separator between the custom step and the custom assembly).

Reply:
Why `,,` ? I think a single comma should be fine. If you are concern about the name being part of type we could force some escaping in that case but don't think that will be needed at this point

Reply:
Basically the only counter-example is something like `MyType,MyAssembly.dll` where on disk there's a `MyAssembly.dll.dll` - so the `.dll` is part of the assembly name and not a file extension.  If we're OK not supporting that case, I'd be perfectly fine with looking at the string after the first comma and if it has a `.dll` or `.exe` extension, then treat it as a file path. To make it even better we could say that in that case there must be just one comma - multiple commas would automatically mean to treat it as an assembly qualified type name.

Reply:
I think we should treat anything after the comma as a file. I don't think allowing assembly-qualified name together with the full path to the file would help anyone. Using the assembly-qualified name without file path would be even stranger and in this context.

Reply:
I was trying to address the fact that linker today supports assembly qualified type names in this context already - so I'm trying to not break that functionality.

Reply:
I don't think that's ever used (and we could do illink support only too)

Reply:
OK - I don't know how important is to keep this for mono linker - so I would vote for changing it everywhere...